pages/_app.js

```diff
@@ -58,15 +58,14 @@ export default function MyApp ({ Component, pageProps: { ...props } }) {
     // So every page load, we modify the route in browser history
     // to point to the same page but without SSR, ie ?nodata=true
     // this nodata var will get passed to the server on back/foward and
-    // 1. prevent data from reloading and 2. perserve scroll
+    // 1. prevent data from reloading and 2. preserve scroll
     // (2) is not possible while intercepting nav with beforePopState
     router.replace({
       pathname: router.pathname,
       query: { ...router.query, nodata: true }
-    }, router.asPath, { ...router.options, shallow: true }).catch((e) => {
+    }, router.asPath, { ...router.options, shallow: false }).catch((e) => {
```

Comment:
This probably has implications elsewhere in the app?

Reply:
yes this will cause the data to load twice on every page visit. no go with this change.